### PR TITLE
[KBFS-1416] Add an "md dump" command to kbfstool

### PR DESCRIPTION
**Pull request by akalin-keybase**
Also rename the kbfs tool to kbfstool.

### CODE REVIEW

File: kbfstool/md_dump.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+	"github.com/keybase/client/go/protocol"
+	"github.com/keybase/kbfs/fsrpc"
+	"github.com/keybase/kbfs/libkbfs"
+	"golang.org/x/net/context"
+)
+
+var mdGetRe = regexp.MustCompile("^(.+?)(?::(.*?))?(?:\\^(.*?))?$")
+
+func getTlfID(
+	ctx context.Context, config libkbfs.Config, tlfPartStr string) (
```
Comment (strib) [inline kbfstool/md_dump.go]: What does the "part" in `tlfPartStr` mean here?  Why not just `tlfIDStr`?

Comment (strib) [inline kbfstool/md_dump.go]: Oh, I guess it can be a handle too.  `tlfIDOrHandleStr`?

File: kbfstool/md_dump.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+	if err != nil {
+		printError("md dump", err)
+		return uid.String()
+	}
+	return fmt.Sprintf("%s (uid:%s)", username, uid)
+}
+
+func mdDumpOne(ctx context.Context, config libkbfs.Config,
+	rmd libkbfs.ImmutableRootMetadata) error {
+	mdID, err := config.Crypto().MakeMdID(&rmd.BareRootMetadata)
```
Comment (strib) [inline kbfstool/md_dump.go]: Why not use `rmd.mdID`?  An immutable root metadata can't be made without one, right?

File: kbfstool/md_dump.go [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+	return nil
+}
+
+const mdDumpUsageStr = `Usage:
+  kbfstool md dump input [inputs...]
+
+Each input must be in the following format:
+
+  TlfPart
```
Comment (strib) [inline kbfstool/md_dump.go]: I don't think you need the "Part" suffixes here either, it makes it more confusing (I keep thinking it means "partial" for some reason).

File: libkbfs/branch_id.go [hunk 4]
```diff
@@ -54,19 +54,20 @@ func (id *BranchID) UnmarshalBinary(data []byte) error {
 	return nil
 }
 
-// ParseBranchID parses a hex encoded BranchID. Returns NullBranchID on failure.
-func ParseBranchID(s string) BranchID {
+// ParseBranchID parses a hex encoded BranchID. Returns NullBranchID
+// and an InvalidBranchID on falire.
+func ParseBranchID(s string) (BranchID, error) {
```
Comment (strib) [inline libkbfs/branch_id.go]: You'll fix up the caller in `kbfs-server`?

**Comment by strib** [comment]
Looks good to me, thanks!

I'm bad at Jenkins, but compiling this is definitely covered by CI, right?

File: kbfstool/md_dump.go (hunk 1 above)
Comment (akalin-keybase) [inline kbfstool/md_dump.go]: Just went with `tlfStr`.

File: kbfstool/md_dump.go (hunk 2 above)
Comment (akalin-keybase) [inline kbfstool/md_dump.go]: Ah, yeah, done! Had to export an accessor for `rmd.mdID`.

File: kbfstool/md_dump.go (hunk 3 above)
Comment (akalin-keybase) [inline kbfstool/md_dump.go]: Done.

File: libkbfs/branch_id.go (hunk 4 above)
Comment (akalin-keybase) [inline libkbfs/branch_id.go]: Yeap!

**Comment by akalin-keybase** [comment]
Yeap, CI covers installing this (since it does `go install kbfs/...`).

**Pull request merged**